modules/volunteers/admin/sitesupport/editPrize.js

```diff
@@ -86,25 +85,65 @@ export default {
       confirmbox('Please! double check entries!', message).then(() => {
-        apiRequest(method, '/volunteer/rewards' + tail, simpleObjectToRequest(item))
-          .then(() => { location.reload();
-          });
+        if (this.reward_group == -2) {
+          let params = [];
+          params.push('reward_limit=' + this.reward_limit);
+          if (this.reward_name) {
+            params.push('name=' + this.reward_name);
+          }
+          apiRequest('POST', '/volunteer/reward_group', params.join('&'))
+            .then((response) => {
+              const data = JSON.parse(response.responseText);
+              item.reward_group = data.id;
+              apiRequest(method, '/volunteer/rewards' + tail, simpleObjectToRequest(item))
+                .then(() => { location.reload(); });
+            });
+        } else {
+          if (this.reward_group > -1 && (
+            (this.reward_limit !== this.$parent.reward_groups[this.reward_group].reward_limit) ||
+            (this.reward_name && this.reward_name !== this.$parent.reward_groups[this.reward_group].name)
+          ))
+          {
+            let params = [];
+            if (this.reward_limit != this.$parent.reward_groups[this.reward_group].reward_limit) {
+              params.push('reward_limit=' + this.reward_limit);
+            }
+            if (this.reward_name != this.$parent.reward_groups[this.reward_group].name) {
+              if (this.reward_name) {
+                params.push('name=' + this.reward_name);
+              }
+            }
+            apiRequest('PUT', '/volunteer/reward_group/' + this.reward_group, params.join('&'))
+              .then(() => {
+                apiRequest(method, '/volunteer/rewards' + tail, simpleObjectToRequest(item))
+                  .then(() => { location.reload(); });
+              });
+          } else {
+            apiRequest(method, '/volunteer/rewards' + tail, simpleObjectToRequest(item))
+              .then(() => { location.reload(); });
+          }
+        }
       });
     },
     updateGroup() {
-      var found = false;
-      this.reward_groups.every((entry) => {
-        if (entry.id == this.reward_group) {
-          found = true;
-          this.reward_limit = entry.reward_limit;
-          return false;
+      this.reward_name = '';
+      if (this.reward_group == -2) {
+        this.reward_limit = 1;
+      } else {
+        if (this.$parent.reward_groups[this.reward_group]) {
+          this.reward_limit = this.$parent.reward_groups[this.reward_group].reward_limit;
+          this.reward_name = this.$parent.reward_groups[this.reward_group].name;
+        } else {
+          this.reward_limit = -1;
         }
-        return true;
-      });
-      if (!found) {
-        this.reward_limit = -1;
       }
     },
+    editPrizeName() {
+      document.getElementById('da_box').style.display = 'inline';
+    },
+    doneEdit() {
+      document.getElementById('da_box').style.display = 'none';
+    }
```

Comment:
Are we unable to get the modal to display through the use of conditional classes/styling here? Perhaps through the use of a boolean on our model, and then apply the class conditionally using that boolean on the modal's definition?

Then we could just update the functions here to set it to true or false based on the appropriate condition (probably `true` for `editPrizeName()` and `false` for `doneEdit()`)

Does something like this not work?

```
this.isEditingPrizeName = false;

<div class="UI-Modal" v-show={{isEditingPrizeName}}>
  ...
</div>
```